astropy/coordinates/transformations.py

```diff
@@ -744,8 +748,190 @@ def __call__(self, fromcoord, toframe):
+        rep = rep.with_differentials(diffs)
+
+        # Only do transform if matrix is specified. This is for speed in
+        # transformations that only specify an offset (e.g., LSR)
+        if matrix is not None:
+            # Note: this applies to both representation and differentials
+            rep = rep.transform(matrix)
+
+        # TODO: if we decide to allow arithmetic between representations that
+        # contain differentials, this can be tidied up
+        if offset is not None:
+            newrep = (rep.without_differentials() +
+                      offset.without_differentials())
+        else:
+            newrep = rep.without_differentials()
+
+        # We need a velocity (time derivative) and, for now, are strict: the
+        # representation can only contain a velocity differential and no others.
+        if has_velocity and not rad_vel_diff:
+            veldiff = rep.differentials['s'] # already in Cartesian form
+
+            if offset is not None and 's' in offset.differentials:
+                veldiff = veldiff + offset.differentials['s']
+
+            newrep = newrep.with_differentials(veldiff)
+
+        if isinstance(fromcoord.data, UnitSphericalRepresentation):
+            # Special-case this because otherwise the return object will think
+            # it has a valid distance with the default return (a
+            # CartesianRepresentation instance)
+
+            if has_velocity and not unit_vel_diff and not rad_vel_diff:
+                # We have to first represent as the Unit types we converted to,
+                # then put the d_distance information back in to the
+                # differentials and re-represent as their original forms
+                newdiff = newrep.differentials['s']
+                _unit_cls = fromcoord.data.differentials['s']._unit_differential
+                newdiff = newdiff.represent_as(_unit_cls, fromcoord.data)
+
+                kwargs = dict([(comp, getattr(newdiff, comp))
+                               for comp in newdiff.components])
+                kwargs['d_distance'] = fromcoord.data.differentials['s'].d_distance
+                diffs = {'s': fromcoord.data.differentials['s'].__class__(
+                    copy=False, **kwargs)}
+
+            else:
+                diffs = newrep.differentials
+
+            newrep = newrep.represent_as(fromcoord.data.__class__) # drops diffs
+            newrep = newrep.with_differentials(diffs)
+
+        # We pulled the radial differential off of the representation
+        # earlier, so now we need to put it back. But, in order to do that, we
+        # have to turn the representation into a repr that is compatible with
+        # having a RadialDifferential
+        if has_velocity and rad_vel_diff:
+            newrep = newrep.represent_as(fromcoord.data.__class__)
+            newrep = newrep.with_differentials({'s': fromcoord.data.differentials['s']})
```

Comment:
This is confusing to me.  If I rotate a `RadialDifferential` it shouldn't be the same differential again, right?  In the extreme case of a 90 degree rotation it should be 0?